sentry-android-core/src/main/java/io/sentry/android/core/SentryLogcatAdapter.java

```diff
@@ -0,0 +1,109 @@
+package io.sentry.android.core;
+
+import android.util.Log;
+import io.sentry.Breadcrumb;
+import io.sentry.Sentry;
+import io.sentry.SentryLevel;
+import org.jetbrains.annotations.ApiStatus;
+import org.jetbrains.annotations.NotNull;
+
+/**
+ * This class replaces {@link android.util.Log} with its own implementations which creates a {@link
+ * io.sentry.Breadcrumb} for each log. It only replaces log functions that meet a minimum level set
+ * by the user on the Sentry Android Gradle Plugin.
+ */
+@ApiStatus.Internal
+public final class SentryLogcatAdapter {
+
+  private static void addAsBreadcrumb(@NotNull String tag, @NotNull SentryLevel level, String msg) {
+    addAsBreadcrumb(tag, level, msg, null);
+  }
+
+  private static void addAsBreadcrumb(
+      @NotNull String tag, @NotNull SentryLevel level, Throwable tr) {
+    addAsBreadcrumb(tag, level, null, tr);
+  }
+
+  private static void addAsBreadcrumb(
+      @NotNull String tag, @NotNull SentryLevel level, String msg, Throwable tr) {
+    Breadcrumb breadcrumb = new Breadcrumb();
+    breadcrumb.setCategory("log");
+    breadcrumb.setMessage(msg);
+    breadcrumb.setLevel(level);
+    breadcrumb.setData("tag", tag);
+    if (tr != null) {
+      breadcrumb.setData("throwable", tr.getMessage());
```

Comment:
I think in the first version it's fine to just have the throwable message inside the breadcrumb, @romtsn what do you think?

Reply:
yeah, I think that's fine